### DISCUSSION TITLE
Supabase Authentication in Appsmith

### DISCUSSION THREAD

**Discussion by inventorado** [supabase/supabase#4666] [General]
Hi

did anyone tried/succeeded in creating Appsmith app that would use Supabase not just for database but would also use Supabase authentication API? My goal is to have appsmith form with email, password and login button and after successful login ability to display data via either SQL or API but only rows that meets RLS policy for connected user. 
I would really appreciate any feedback as I'm a bit struggling here after few hours of googling and tries and errors. Samples I found only uses supabase database in appsmith, but not authentication and user management. Thank you very much!

**Comment by inventorado** [comment]
OK, I figured it out

**Reply by Brutus5000** [reply]
Please share your insights with us. Googling your problem brought me here.

**Reply by gaa23** [reply]
Yeah I would love to see how you did it too.

**Comment by GreenFlux** [comment]
Old thread but I wanted to post the solution for anyone else that finds this post. 

I was able to get this working with Supabase email login, and the data is filtered by RLS based on the user. Here's how to set it up on the Appsmith side. 

1. Add input widgets for the user name and password
2. Create a new API: _(copy, then import with cURL in Appsmith)_
```
curl -XPOST -H 'apikey: APIKEY' -H "Content-type: application/json" -d '{
  "email": {{in_loginEmail.text}},
  "password": {{in_loginPassword.text}}
}' 'https://PROJECT_ID.supabase.co/auth/v1/token?grant_type=password'
```
- Add your API key and project ID

3. Add a new JS Object and write a function to save the API response to the Appsmith store
```
	supaLoginHelper: async () => {
		return await supaLogin.run( async data=>{await storeValue('login',data);navigateTo('Profile')}, e=>showAlert(e.error_description) )
	},
```

4. Add the `access_token` to the header of any APIs that require authentication:
```
Authorization: Bearer {{appsmith.store.login.access_token}}
```

I have a `user_id` field in each table that autofills with `auth.uid()`, and I'm using `auth.uid() = user_id` for the RLS condition. So each user can only perform CRUD ops on their own records.

**Comment by patrickswelsh** [comment]
Hey - wanted to give another example here - [this is a video](https://www.youtube.com/watch?v=mfhHUDNCkoQ&t=211s) from the people at Appsmith on the topic, its very detailed and covers a few more instances (social logins, creating new accounts).

Also - in case this wasn't obvious: to avoid your users having to log in twice (once to appsmith, and again to supabase), you need to [make your appsmith project public](https://docs.appsmith.com/advanced-concepts/access-control), and rely on the supabase auth to protect your app. You'll have to use a self-hosted version of Appsmith to do this... you have to be an administrator to make a project public. There's great docs on setting up for self-hosted on Appsmith's site.

**Reply by eedwardsSGS** [reply]
Wanted to tack on since I found this thread through Google:
The video link address is broken but if you replace it with Youtube it works https://www.youtube.com/watch?v=mfhHUDNCkoQ&t=211s

Also Appsmith now allows importing the Supabase-JS library!  (But I couldn't figure out how to implement the password reset functions using it.)
